src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1278,22 +1276,21 @@ static ImmutableArray<IArgumentOperation> getCreationArguments(
                 CSharpOperationFactory @this, BoundCollectionExpression expr)
             {
                 var collectionCreation = expr.CollectionCreation;
-                if (collectionCreation is { HasAnyErrors: true })
-                    return [];
-
                 while (collectionCreation is BoundConversion conversion)
                     collectionCreation = conversion.Operand;
 
                 if (collectionCreation is BoundObjectCreationExpression)
                     return @this.DeriveArguments(collectionCreation);
 
-                if (collectionCreation is BoundCall)
+                // Match the logic in CreateBoundCallOperation which does not DeriveArguments in the case of an
+                // erroneous call node.
+                if (collectionCreation is BoundCall { IsErroneousNode: false })
```

Comment:
We still need to have the arguments in the tree, even in this case. We can just use `IInvalidOperation` for them.

Reply:
> Perhaps we loosen the node type then? And document that it's either an `IArgumentOperation` or `IInvalidOperation`? I'd like thoughts from @AlekseyTs as well.

I don't think it make sense to create an invalid operation explicitly. How does the following sound:
- We produce ```ImmutableArray<IOperation>```   
- When binding is successful, items are ```IArgumentOperation```s
- When binding fails, items are some other ```IOperations``` representing nodes corresponding to the argument expressions in syntax